diff -ur funkwhale-1.2.1.orig/api/funkwhale_api/users/models.py funkwhale-1.2.1/api/funkwhale_api/users/models.py
--- funkwhale-1.2.1.orig/api/funkwhale_api/users/models.py	2022-03-04 17:28:45.644000292 -0800
+++ funkwhale-1.2.1/api/funkwhale_api/users/models.py	2022-03-04 20:29:45.000001889 -0800
@@ -17,7 +17,6 @@
 from django.utils.translation import ugettext_lazy as _
 
 from allauth.account.models import EmailAddress
-from django_auth_ldap.backend import populate_user as ldap_populate_user
 from oauth2_provider import models as oauth2_models
 from oauth2_provider import validators as oauth2_validators
 from versatileimagefield.fields import VersatileImageField
@@ -453,9 +452,3 @@
     args["public_key"] = public.decode("utf-8")
 
     return federation_models.Actor.objects.create(user=user, **args)
-
-
-@receiver(ldap_populate_user)
-def init_ldap_user(sender, user, ldap_user, **kwargs):
-    if not user.actor:
-        user.actor = create_actor(user)
diff -ur funkwhale-1.2.1.orig/api/requirements/base.txt funkwhale-1.2.1/api/requirements/base.txt
--- funkwhale-1.2.1.orig/api/requirements/base.txt	2022-03-04 17:28:45.644000292 -0800
+++ funkwhale-1.2.1/api/requirements/base.txt	2022-03-04 20:29:59.304001903 -0800
@@ -49,10 +49,6 @@
 requests~=2.26.0
 pyOpenSSL~=20.0.1
 
-# for LDAP authentication
-python-ldap~=3.3.0
-django-auth-ldap~=3.0.0
-
 pydub~=0.25.1
 pyld~=2.0.3
 aiohttp~=3.7.4
