diff -ur funkwhale-1.3.0.orig/api/funkwhale_api/users/models.py funkwhale-1.3.0/api/funkwhale_api/users/models.py
--- funkwhale-1.3.0.orig/api/funkwhale_api/users/models.py	2023-06-01 16:38:50.000000000 +0000
+++ funkwhale-1.3.0/api/funkwhale_api/users/models.py	2023-06-01 20:34:23.998772945 +0000
@@ -13,7 +13,6 @@
 from django.urls import reverse
 from django.utils import timezone
 from django.utils.translation import ugettext_lazy as _
-from django_auth_ldap.backend import populate_user as ldap_populate_user
 from oauth2_provider import models as oauth2_models
 from oauth2_provider import validators as oauth2_validators
 from versatileimagefield.fields import VersatileImageField
@@ -457,9 +456,3 @@
     args["public_key"] = public.decode("utf-8")
 
     return federation_models.Actor.objects.create(user=user, **args)
-
-
-@receiver(ldap_populate_user)
-def init_ldap_user(sender, user, ldap_user, **kwargs):
-    if not user.actor:
-        user.actor = create_actor(user)
diff -ur funkwhale-1.3.0.orig/api/pyproject.toml funkwhale-1.3.0/api/pyproject.toml
--- funkwhale-1.3.0.orig/api/pyproject.toml	2023-06-01 16:38:50.000000000 +0000
+++ funkwhale-1.3.0/api/pyproject.toml	2023-06-01 20:39:58.779908925 +0000
@@ -49,10 +49,6 @@
 psycopg2 = "==2.9.5"
 redis = "==4.5.4"
 
-# Django LDAP
-django-auth-ldap = "==4.1.0"
-python-ldap = "==3.4.3"
-
 # Channels
 channels = { extras = ["daphne"], version = "==4.0.0" }
 channels-redis = "==4.1.0"
