diff -ur funkwhale-1.2.1.orig/deploy/funkwhale-server.service funkwhale-1.2.1/deploy/funkwhale-server.service
--- funkwhale-1.2.1.orig/deploy/funkwhale-server.service	2022-01-28 16:39:20.971946198 -0800
+++ funkwhale-1.2.1/deploy/funkwhale-server.service	2022-01-28 16:42:07.015943294 -0800
@@ -4,5 +4,6 @@
 PartOf=funkwhale.target
 
 [Service]
+Type=notify
 User=funkwhale
 # adapt this depending on the path of your funkwhale installation
